src/Symfony/Component/EventDispatcher/LegacyEventDispatcherProxy.php

```diff
@@ -16,14 +16,16 @@
 /**
  * A helper class to provide BC/FC with the legacy signature of EventDispatcherInterface::dispatch().
  *
- * This class should be deprecated in Symfony 5.1
- *
  * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @deprecated since Symfony 5.1.
  */
 final class LegacyEventDispatcherProxy
 {
     public static function decorate(?EventDispatcherInterface $dispatcher): ?EventDispatcherInterface
     {
+        @trigger_error(sprintf('%s is deprecated. Use the event dispatcher without the proxy.', __CLASS__), E_USER_DEPRECATED);
```

Comment:
can we put this below the namespace ?

Reply:
But then the `class_exists()` call I've used in the notifier component would trigger the deprecation, wouldn't it?

Reply:
ok, got it
maybe replace with `class_exists(Symfony\Component\EventDispatcher\Event::class)` everywhere?

Reply:
Good idea.